Add SECURITY.md to the repo (#5929)
Cog-Creators/Red-DiscordBot: SECURITY.md

# Security Policy

## Supported Versions

> The table below explains the current state of our versions. Currently only version

Reviewer: ```suggestion
The table below explains the current state of our versions. Currently, only version
```

are considered End of Life and will not receive any security updates.

| Version       | Branch     | Security Updates   | End of Life        |
|---------------|------------|--------------------|--------------------|
| < 2.0         | master     | :x:                | :white_check_mark: |
| >= 2.0, < 3.0 | develop    | :x:                | :white_check_mark: |
| >= 3.0, < 3.4 | V3/develop | :x:                | :white_check_mark: |
| >= 3.4        | V3/develop | :white_check_mark: | :x:                |

## Reporting a Vulnerability

For reporting vulnerabilities within Red-DiscordBot we make use of GitHub's
private vulnerability reporting feature (More information can be found
[here](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability)).
This ensures that all maintainers and key members have access to the reported
vulnerability.

### Opening a Vulnerability Report

To open a vulnerability report please fill out [this form](https://github.com/Cog-Creators/Red-DiscordBot/security/advisories/new)

You will be asked to provide a summary, details and proof of concept for your vulnerability report.

> We ask that you fill out these details to the best of your ability, with as many details as possible.

Reviewer: ```suggestion
We ask that you fill out this form to the best of your ability, with as many details as possible.
```

> Furthermore you'll be asked to provide affected products and severity. These are optional to fill out.
> And will be handled by the maintainers.

Reviewer: The sentence split on an 'and' feels weird, maybe this is better?
```suggestion
Furthermore, you'll be asked to provide affected products and severity.
These fields are optional and will be filled appropriately by the maintainers if not provided.
```

### Timeline

> We will try to answer your report within 3 days. If you haven't received a answer we suggest you reach

Reviewer: ```suggestion
We will try to answer your report within 3 days. If you haven't received an answer by then, we suggest you reach
```

> out to us privately. This can best be done via our [Discord server](https://discord.gg/red).

Reviewer: I'm not sure if this is specific enough, it's a big server and bad actors can be a risk if say, the reporter asks in a public channel who they can contact about a vulnerability.

Author: Added some more specific text there. Key reason for staff marked is that they'll forward it to the right people.
